db/nft.go

```diff
@@ -426,24 +443,28 @@ func GetCreators(conn *pgxpool.Conn, q QueryCreatorRequest, p PageRequest) (res
 			'count', count)),
 		COUNT(*) OVER() AS row_count
 	FROM (
-		SELECT i.owner, i.iscn_id_prefix, c.class_id, SUM(%s.latest_price) AS value, COUNT(DISTINCT n.id) as count
+		SELECT i.owner, i.iscn_id_prefix, c.class_id, SUM(%[1]s) AS value, COUNT(DISTINCT n.id) as count
 		FROM iscn AS i
 		JOIN iscn_latest_version
 		ON i.iscn_id_prefix = iscn_latest_version.iscn_id_prefix
 			AND ($5 = true OR i.version = iscn_latest_version.latest_version)
 		JOIN nft_class AS c ON i.iscn_id_prefix = c.parent_iscn_id_prefix
 		JOIN nft AS n ON c.class_id = n.class_id
 			AND ($4::text[] IS NULL OR cardinality($4::text[]) = 0 OR n.owner != ALL($4))
+		JOIN nft_event AS e 
+		ON e.nft_id = n.nft_id
+			AND e.receiver = n.owner
 		WHERE 
 			($6 = true OR n.owner != i.owner)
 			AND ($1::text[] IS NULL OR cardinality($1::text[]) = 0 OR n.owner = ANY($1))
+			AND e.price IS NOT NULL
```

Comment:
Should we add event filter by action type? Not sure if in both buy and sell event `receiver` has the same meaning

Reply:
Seems the `receiver` of both `buy_nft` and `sell_nft` means the one who eventually has the NFT.

This can be verified from tests `TestListing()` and `TestOffer()` in [marketplace_test.go](https://github.com/WeiJun0827/likecoin-chain-tx-indexer/blob/feature/creator-collector/extractor/marketplace_test.go).

In both cases, `ADDR_02_LIKE` (`like1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqm5jeaq`) buys an NFT from `ADDR_01_LIKE` (`like1qyqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqewmlu9`) in the last tx (`"txhash":"AAAAAE"`).
And the `receiver` field is `ADDR_02_LIKE` in both results.
<img width="1481" alt="截圖 2023-03-25 上午10 54 24" src="https://user-images.githubusercontent.com/33746295/227687290-c15ba6a1-73ce-46fe-a644-a7c2e9b9db54.png">
<img width="1481" alt="截圖 2023-03-25 上午10 56 01" src="https://user-images.githubusercontent.com/33746295/227687339-4f82234c-17c4-42bf-839c-0acb3175b11e.png">

Should we still need an event filter here?

Reply:
It would be fine then, any comment from @nnkken ?